zoomable/src/commonMain/kotlin/me/saket/telephoto/zoomable/internal/keyboardActions.kt

```diff
@@ -0,0 +1,139 @@
+package me.saket.telephoto.zoomable.internal
+
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.geometry.Offset
+import androidx.compose.ui.input.key.Key
+import androidx.compose.ui.input.key.KeyEvent
+import androidx.compose.ui.input.key.KeyEventType
+import androidx.compose.ui.input.key.KeyInputModifierNode
+import androidx.compose.ui.input.key.isCtrlPressed
+import androidx.compose.ui.input.key.isMetaPressed
+import androidx.compose.ui.input.key.key
+import androidx.compose.ui.input.key.type
+import androidx.compose.ui.node.ModifierNodeElement
+import androidx.compose.ui.unit.Density
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.DpOffset
+import androidx.compose.ui.unit.dp
+
+/**
+ * Responds to keyboard events to zoom and pan
+ */
+internal data class KeyboardActionsElement(
+  private var zoomStep: Float,
+  private var panStep: Dp,
+  private var canPan: () -> Boolean,
+  private var onZoom: (Float) -> Unit,
+  private var onResetZoom: () -> Unit,
+  private var onPan: (DpOffset) -> Unit,
+) : ModifierNodeElement<KeyboardActionsNode>() {
+  override fun create(): KeyboardActionsNode {
+    return KeyboardActionsNode(
+      zoomStep = zoomStep,
+      panStep = panStep,
+      canPan = canPan,
+      onZoom = onZoom,
+      onResetZoom = onResetZoom,
+      onPan = onPan,
+    )
+  }
+
+  override fun update(node: KeyboardActionsNode) {
+    node.update(
+      zoomStep = zoomStep,
+      panStep = panStep,
+      canPan = canPan,
+      onZoom = onZoom,
+      onResetZoom = onResetZoom,
+      onPan = onPan,
+    )
+  }
+}
+
+internal class KeyboardActionsNode(
+  private var zoomStep: Float,
+  private var panStep: Dp,
+  private var canPan: () -> Boolean,
+  private var onZoom: (Float) -> Unit,
+  private var onResetZoom: () -> Unit,
+  private var onPan: (DpOffset) -> Unit,
+) : Modifier.Node(), KeyInputModifierNode {
+  override fun onKeyEvent(event: KeyEvent): Boolean {
+    //TODO: how to detect the correct key combos?
+    println("onKeyEvent: ${event.key} isCtrlPressed=${event.isCtrlPressed} isMetaPressed:${event.isMetaPressed}")
+    when {
+      event.key == Key.ZoomIn || event.key == Key.Plus || event.key == Key.Equals -> {
+        if (event.type == KeyEventType.KeyDown) {
+          onZoom(zoomStep)
+        }
+        return true
+      }
+      event.key == Key.ZoomOut || event.key == Key.Minus -> {
+        if (event.type == KeyEventType.KeyDown) {
+          onZoom(1 / zoomStep)
+        }
+        return true
+      }
+      event.key == Key.Zero -> {
```

Comment:
I'm curious, what was the reason for supporting `Key.Zero`? I haven't been able to find any other app/OS that resets zoom when the `zero` key is pressed.

Reply:
Because my browser does, didn't think too much about it and don't have strong opinions